android/canonical/app/src/main/java/com/google/samples/quickstart/canonical/ProfileFragment.kt

```diff
@@ -0,0 +1,72 @@
+package com.google.samples.quickstart.canonical
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ListView
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.Observer
+import androidx.databinding.DataBindingUtil
+import com.google.samples.quickstart.canonical.databinding.FragmentProfileBinding
+
+
+class ProfileFragment : Fragment() {
+
+    private val signInVM: SignInViewModel by activityViewModels()
+    private val profileVM : ProfileViewModel by activityViewModels()
+    private lateinit var binding : FragmentProfileBinding
+
+    private fun downloadPhotoAndSetView(view: View) {
+        val url = profileVM.getUserPhotoURL()
+        if (url != "") {
+            Log.d(PROFILE_TAG, url)
+            DownloadImageTask(view.findViewById(R.id.usr_img))
+                .execute(url)
+        }
+    }
+
+    private fun setRunHistory() {
+        val runHistoryListForView : ArrayList<ProfileViewModel.SingleRun>
+                = profileVM.getRunHistoryListForView()
+        val runHistoryAdapter = RunHistoryAdapter(requireContext(), runHistoryListForView)
+        val runHistoryListView = view?.findViewById<ListView>(R.id.run_history_list_view)
```

Comment:
RecyclerView is the modern replacement for ListView. For this case I think ListView is fine. https://developer.android.com/jetpack/androidx/releases/recyclerview?hl=en